### PR TITLE
Add FenwickTree for Python

### LINKED ISSUE #220: [Data Structure] Fenwick Tree [Python]
The FenwickTree is a data structure for O(log(n)) updates and queries that answers both point-update and range-query questions, find out more here: https://en.wikipedia.org/wiki/Fenwick_tree
It would be nice to have this data structure implemented in Python.

### PR DESCRIPTION
**Pull request by SnowyCoder**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

Closes #220 

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

File: fenwick_tree/python/fenwicktree.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    # Set initial value(by default it's 0)
+    for i in range(8):
+        tree.add(i, i)
+
+    # Now it's 0, 1, 2, 3, 4, 5, 6, 7
+
+    # [0, 1, 2, 3, 4, 5, 6, 7]
+    #  ^--------------------^
+    # 0 + 1 + 2...+ 7 = 28
+    assert tree.sum(0, 7) == 28;
```
Comment (sangamcse) [inline fenwick_tree/python/fenwicktree.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpe4vx7obs/fenwick_tree/python/fenwicktree.py
+++ b/tmp/tmpe4vx7obs/fenwick_tree/python/fenwicktree.py
@@ -62,7 +62,7 @@
     # [0, 1, 2, 3, 4, 5, 6, 7]
     #  ^--------------------^
     # 0 + 1 + 2...+ 7 = 28
-    assert tree.sum(0, 7) == 28;
+    assert tree.sum(0, 7) == 28
 
     # [0, 1, 2, 3, 4, 5, 6, 7]
     #  ^-----^
```

File: fenwick_tree/python/fenwicktree.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    # [0, 1, 2, 3, 4, 5, 6, 7]
+    #  ^-----^
+    #  4 + 5 + 6 = 15
+    assert tree.sum(4, 6) == 15;
```
Comment (sangamcse) [inline fenwick_tree/python/fenwicktree.py]: The code does not comply to PEP8.

*Origin: PEP8Bear, Section: `all.autopep8`.*

The issue can be fixed by applying the following patch:

```diff
--- a/tmp/tmpe4vx7obs/fenwick_tree/python/fenwicktree.py
+++ b/tmp/tmpe4vx7obs/fenwick_tree/python/fenwicktree.py
@@ -67,7 +67,7 @@
     # [0, 1, 2, 3, 4, 5, 6, 7]
     #  ^-----^
     #  4 + 5 + 6 = 15
-    assert tree.sum(4, 6) == 15;
+    assert tree.sum(4, 6) == 15
 
     # [0, 1, 2, 3, 4, 5, 6, 7]
     #  ^--------^
```

File: fenwick_tree/python/fenwicktree.py (hunk 1 above)
Comment (sangamcse) [inline fenwick_tree/python/fenwicktree.py]: E703 statement ends with a semicolon

*Origin: PycodestyleBear (E703), Section: `all.autopep8`.*

File: fenwick_tree/python/fenwicktree.py (hunk 2 above)
Comment (sangamcse) [inline fenwick_tree/python/fenwicktree.py]: E703 statement ends with a semicolon

*Origin: PycodestyleBear (E703), Section: `all.autopep8`.*

**Comment by SnowyCoder** [comment]
Sorry I missed it.
I'll add it once one of the other fenwick tree implementations are merged (to avoid merge issues).

File: fenwick_tree/python/fenwick_tree.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+
+
```
Comment (sangamcse) [inline fenwick_tree/python/fenwick_tree.py]: can you remove these 2 blank lines?

**Comment by sangamcse** [comment]
@SnowyCoder, I couldn't see `Java` as `java` and that was my mistake in your previous PR. Sorry for that. I will create an issue for that later on.

In addition to the above review, can you correct the commit message also as the Header of the commit should be 50 characters and body should be 72 characters per line? So you need to break line after 72 characters.

**Comment by SnowyCoder** [comment]
Yep, I'm changing it

**Comment by SnowyCoder** [comment]
Sorry for the trouble

**Comment by sangamcse** [comment]
> Sorry for the trouble

Don't be sorry 😄

**Comment by sangamcse** [comment]
ack a0d06d5

**Pull request merged**